packages/web-components/fast-foundation/src/data-grid/data-grid-row.ts

```diff
@@ -0,0 +1,293 @@
+        this.removeEventListener("keydown", this.handleKeydown);
+    }
+
+    public handleFocusout(e: FocusEvent): void {
+        if (!this.contains(e.target as Element)) {
+            this.isActiveRow = false;
+            this.focusColumnIndex = 0;
+        }
+    }
+
+    public handleCellFocus(e: Event): void {
+        this.isActiveRow = true;
+        const cells: Element[] = Array.from(
+            this.querySelectorAll(DataGridRow.cellQueryString)
+        );
+        this.focusColumnIndex = cells.indexOf(e.target as Element);
+        this.$emit("row-focused", this);
+    }
+
+    public handleKeydown(e: KeyboardEvent): void {
+        if (e.defaultPrevented) {
+            return;
+        }
+        let cells: Element[] = [];
+        let newFocusColumnIndex: number = 0;
+        switch (e.keyCode) {
+            case keyCodeArrowLeft:
+                // focus left one cell
+                cells = Array.from(this.querySelectorAll(DataGridRow.cellQueryString));
+                newFocusColumnIndex = Math.max(0, this.focusColumnIndex - 1);
+                (cells[newFocusColumnIndex] as HTMLElement).focus();
+                e.preventDefault();
+                break;
+
+            case keyCodeArrowRight:
+                // focus right one cell
+                cells = Array.from(this.querySelectorAll(DataGridRow.cellQueryString));
+                newFocusColumnIndex = Math.min(
+                    cells.length - 1,
+                    this.focusColumnIndex + 1
+                );
+                (cells[newFocusColumnIndex] as HTMLElement).focus();
+                e.preventDefault();
+                break;
+
+            case keyCodeHome:
+                if (!e.ctrlKey) {
+                    // focus first cell of the row
+                    cells = Array.from(
+                        this.querySelectorAll(DataGridRow.cellQueryString)
+                    );
+                    (cells[0] as HTMLElement).focus();
+                    e.preventDefault();
+                }
+                break;
+            case keyCodeEnd:
+                if (!e.ctrlKey) {
+                    // focus last cell of the row
+                    cells = Array.from(
+                        this.querySelectorAll(DataGridRow.cellQueryString)
+                    );
+                    (cells[cells.length - 1] as HTMLElement).focus();
+                    e.preventDefault();
+                }
+                break;
+        }
+    }
+
+    private updateItemTemplate(): void {
+        this.activeCellItemTemplate =
+            this.rowType === DataGridRowTypes.default
+                ? this.cellItemTemplate !== undefined
+                    ? this.cellItemTemplate
+                    : this.defaultCellItemTemplate
+                : this.headerCellItemTemplate !== undefined
+                ? this.headerCellItemTemplate
+                : this.defaultHeaderCellItemTemplate;
```

Comment:
I'm a bit torn here, because while I think nested ternarys are very difficult to parse, I also think you end up doing additional checks. I prefer the clarity of removing the nested ternarys though from a maintenance perspective.

```suggestion
            this.rowType === DataGridRowTypes.default && this.cellItemTemplate !== undefined
                ? this.cellItemTemplate
                : this.rowType === DataGridRowTypes.default && this.cellItemTemplate === undefined
                ? this.defaultCellItemTemplate
                : this.headerCellItemTemplate !== undefined
                ? this.headerCellItemTemplate
                : this.defaultHeaderCellItemTemplate;
```